### ISSUE TITLE
"Object [object Undefined] has no method 'trim'"  on "local push"

### ISSUE THREAD

**Issue by pimterry** [resin-io/resin-cli#545]
https://sentry.io/resinio/cli/issues/287407676/

```
TypeError: Object [object Undefined] has no method 'trim'
  File "/usr/lib/node_modules/resin-cli/node_modules/resin-cli-form/node_modules/bluebird/js/main/call_get.js", line 75, in ensureMethod
    throw new Promise.TypeError(message);
  File "<anonymous>", line 7, in Array.eval (eval at makeMethodCaller (/usr/lib/node_modules/resin-cli/node_modules/resin-cli-form/node_modules/bluebird/js/main/call_get.js:18:12),
  File "/usr/lib/node_modules/resin-cli/node_modules/resin-cli-form/node_modules/bluebird/js/main/util.js", line 26, in Array.tryCatcher
    return target.apply(this, arguments);
  File "/usr/lib/node_modules/resin-cli/node_modules/resin-cli-form/node_modules/bluebird/js/main/promise.js", line 510, in Promise._settlePromiseFromHandler
    x = tryCatch(handler).call(receiver, value);
  File "/usr/lib/node_modules/resin-cli/node_modules/resin-cli-form/node_modules/bluebird/js/main/promise.js", line 584, in Promise._settlePromiseAt
    this._settlePromiseFromHandler(handler, receiver, value, promise);
...
(7 additional frame(s) were not displayed)

TypeError: Object [object Undefined] has no method 'trim'
```

The full command being used is `resin local push 36fa33e.local -s` - I think this is missing a part argument after the `-s`, so throwing. We should handle this (and cases like this generally) more gracefullly.

**Comment by pimterry** [comment]
Another similar one: https://sentry.io/resinio/cli/issues/319528496/

**Comment by pimterry** [comment]
https://sentry.io/resinio/cli/issues/253238062/

**Comment by pimterry** [comment]
Another one: https://sentry.io/resinio/cli/issues/354461153/

I believe this happens when it asks for an app name, if you just don't enter anything.

Short-term we should fix this by removing manual local push naming entirely. This is indicative of a capitano (or whatever's building the form here) bug though: if you enter nothing that should return `""`, not undefined (and even better, it should validate against that).

**Comment by pimterry** [comment]
This has now been fixed in resin-sync 9.2.4, which removes the container name picking functionality completely to fix #682. That should get picked up automatically by all future CLI installs.
